pkg/providers/instanceprofile/instanceprofile.go

```diff
@@ -129,5 +148,55 @@ func (p *DefaultProvider) Delete(ctx context.Context, instanceProfileName string
 		return awserrors.IgnoreNotFound(serrors.Wrap(fmt.Errorf("deleting instance profile, %w", err), "instance-profile", instanceProfileName))
 	}
 	p.cache.Delete(instanceProfileName)
+	p.protectedProfiles.Delete(instanceProfileName)
```

Comment:
nit: let's just use `SetProtectedState` here rather than updating the cache directly.